core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -952,10 +971,8 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
   private Iterable<ManifestFile> prepareNewDataManifests() {
     Iterable<ManifestFile> newManifests;
     if (newDataFiles.size() > 0) {
-      ManifestFile newManifest = newDataFilesAsManifest();
-      newManifests =
-          Iterables.concat(
-              ImmutableList.of(newManifest), appendManifests, rewrittenAppendManifests);
+      List<ManifestFile> newManifest = newDataFilesAsManifests();
```

Comment:
Should this be called something like `dataFileManifests` given that it is a list now?

Reply:
Yeah, updated.